src/InterventionBackend.php

```diff
@@ -1,54 +0,0 @@
-<?php
```

Comment:
Invalidating cache with params to `variantName` seems to work fine already. I'm not too sure if overriding the default image-manipulation methods is actually a good idea though? What if I want the original way to crop an image? I think having specific method names is better to be honest…

We could also integrate some of the methods on the `DBFocusPoint` field. So it could  have a `Fill` method and you would then invoke it via `$Image.FocusPoint.Fill`… still more to write than `$Image.FocusFill` though ;)

Reply:
> Invalidating cache with params to variantName seems to work fine already.

Yep it works fine for FocusFill, but not if you're replacing the built in methods. Replacing the built in methods is something I'm very keen on, but it could be optional. I would love FocusPoint to be a drop in replacement for cropping and can't think of any instance where a centred crop would be better than a focused one. This would mean things like CMS thumbnails benefit from focus based cropping and any themes or templates that use built in cropping methods wouldn't need to be overridden to get upgraded cropping.

Reply:
I haven't tested yet but I understand a trait works as if the code in it has been copy-pasted in to whatever is using it. So if I put an extension point in that method I should be able to apply an extension to Image and DBFile (the two classes that use it) to hook in to it.